pkg/apis/pipeline/v1/pipeline_types.go

```diff
@@ -212,6 +212,24 @@ type Matrix struct {
 	// The names of the `params` in the `Matrix` must match the names of the `params` in the underlying `Task` that they will be substituting.
 	// +listType=atomic
 	Params []Param `json:"params,omitempty"`
+
+	// Include is a list of MatrixInclude which allows passing in specific combinations of Parameters into the Matrix.
+	// Note that Include is in preview mode and not yet supported.
+	// +optional
+	// +listType=atomic
+	Include []MatrixInclude `json:"include,omitempty"`
+}
+
+// MatrixInclude allows passing in a specific combinations of Parameters into the Matrix.
+// Note this struct is in preview mode and not yet supported
+type MatrixInclude struct {
+	// Name the specified combination
+	Name string `json:"name,omitempty"`
+
+	// Params takes only `Parameters` of type `"string"`
```

Comment:
planning to add validation that the parameters are of type string in a subsequent PR?

Reply:
Updated commit message to call out that validation will be added in a subsequent PR.